src/vulkan-renderer/render_graph.cpp

```diff
@@ -519,39 +534,29 @@ VkSemaphore RenderGraph::render(std::uint32_t image_index, VkSemaphore wait_sema
         }
     }
 
-    // Re-record all command buffers. This isn't great, but it's fine for simple rendering pipelines. Eventually we'll
-    // want to generate these in parallel.
-    vkResetCommandPool(m_device.device(), m_command_pool, 0);
-    for (const auto *stage : m_stage_stack) {
-        record_command_buffer(stage, *stage->m_physical, image_index);
-    }
+    const std::array<VkPipelineStageFlags, 1> stage_mask{VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT};
+    std::vector<VkSubmitInfo> submit_infos;
 
-    auto submit_info = wrapper::make_info<VkSubmitInfo>();
-    submit_info.commandBufferCount = 1;
-    submit_info.signalSemaphoreCount = 1;
-    submit_info.waitSemaphoreCount = 1;
+    for (const auto &stage : m_stage_stack) {
+        const auto &cmd_buf = m_device.request_command_buffer("rendergraph");
```

Comment:
Can be moved outside of the loop as well as making a single submit info.

Reply:
Done.